src/merkle/merkle-tree.ts

```diff
@@ -50,7 +50,7 @@ export class MerkleTree<T> {
       throw new Error('Cannot generate Merkle structure with no elements');
     }
 
-    if (this.depthLimit && treeDepth > this.depthLimit) {
+    if (this.depthLimit > 0 && treeDepth > this.depthLimit) {
```

Comment:
Any chance that the depthLimit was set to `'0'` (string) or something like that?

Reply:
Yeah that is likely

Reply:
man I hate javascript 😅

Reply:
So... the auto-generated `Config.d.ts` file does have the `merkleDepthLimit` options specified to be of type `number`... but maybe at runtime that isn't actually what happens?

Reply:
sure enough. I logged `typeof config.merkleDepthLimit` and it's a string.  Sigh